File: VB/MvcApp_Q398100/Models/Northwind.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Collections
Imports DevExpress.Web.ASPxGridView
Imports MvcApp_Q398100.Models

Public NotInheritable Class NorthwindDataProvider
	Private Shared db_Renamed As NorthwindDataContext
	Private Sub New()
	End Sub
	Public Shared ReadOnly Property DB() As NorthwindDataContext
		Get
			If db_Renamed Is Nothing Then
				db_Renamed = New NorthwindDataContext()
			End If
			Return db_Renamed
		End Get
	End Property
	Public Shared Function GetProducts() As IEnumerable
		Return _
			From product In DB.Products _
			Select product
	End Function
End Class

File: CS/MvcApp_Q398100/Models/Northwind.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Collections;
using DevExpress.Web;
using MvcApp_Q398100.Models;

public static class NorthwindDataProvider {
    static NorthwindDataContext db;
    public static NorthwindDataContext DB {
        get {
            if (db == null)
                db = new NorthwindDataContext();
            return db;
        }
    }
    public static IEnumerable GetProducts() {
        return from product in DB.Products select product;
    }
}
